src/p2p/net_node.inl

```diff
@@ -2311,7 +2335,7 @@ namespace nodetool
     }
     network_zone& zone = m_network_zones.at(na.get_zone());
 
-    if(!zone.m_peerlist.is_host_allowed(context.m_remote_address))
+    if(!zone.m_peerlist.is_host_allowed(na))
```

Comment:
I think this might be abusable by making lots of connections to a node, embedding a return IP of a target host, and you get to spam the host, shielding the real origin from the target. Not sure whether that's a viable attack though, since it doesn't amplify anything.